lib/m2ts/caption-stream.js

```diff
@@ -99,65 +99,57 @@ CaptionStream.prototype.push = function(event) {
   this.latestDts_ = event.dts;
 };
 
-CaptionStream.prototype.sortAndDispatchPackets = function(packets) {
+CaptionStream.prototype.flushStream = function(flushType) {
+  if (flushType !== 'flush' && flushType !== 'partialFlush') {
+    return;
+  }
+
+  var flushCCStreams = function(captionStream) {
```

Comment:
Probably better to pull this function out and take a parameter for flush type. Less nesting, shorter functions, and can be tested separately if we want.